core/src/main/java/software/amazon/awssdk/runtime/StandardMemberCopier.java

```diff
@@ -74,6 +74,19 @@ public static ByteBuffer copy(ByteBuffer bb) {
         if (bb == null) {
             return null;
         }
-        return bb.duplicate();
+
+        ByteBuffer copy = bb.isDirect() ? ByteBuffer.allocateDirect(bb.capacity()) : ByteBuffer.allocate(bb.capacity());
```

Comment:
Not really sure, I feel like a heap allocated byte buffer would be fine since I don't think we would get the benefits of direct if we have to copy it. The docs say direct has higher allocation/deallocation costs than non-direct and our version will probably be more short lived (unless they reuse the request object). 

"A direct byte buffer may be created by invoking the allocateDirect factory method of this class. The buffers returned by this method typically have somewhat higher allocation and deallocation costs than non-direct buffers. The contents of direct buffers may reside outside of the normal garbage-collected heap, and so their impact upon the memory footprint of an application might not be obvious. It is therefore recommended that direct buffers be allocated primarily for large, long-lived buffers that are subject to the underlying system's native I/O operations. In general it is best to allocate direct buffers only when they yield a measureable gain in program performance."